core/src/main/java/jenkins/security/s2m/AdminCallableMonitor.java

```diff
@@ -53,6 +53,7 @@ public AdminWhitelistRule getRule() {
      */
     @RequirePOST
     public HttpResponse doAct(@QueryParameter String dismiss) throws IOException {
+        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
```

Comment:
I'll take another look at this 👍

Reply:
I've added this to the disable call:
https://github.com/jenkinsci/jenkins/pull/4149/files#diff-eb8d855c59ff780930c24370849acef6R119

If the only action that is taken is disable then the admin monitor doesn't need an update after this change

but do you have any ideas of how to handle arbitrary jelly code and what it calls?

I've updated the admin monitors I was able to trigger so far

Reply:
That doesn't really help. Admin monitors may have further actions (`doAct` seems popular), and they haven't needed individual permission checks for a while because the `StaplerProxy` approach shut that down completely.

Jelly isn't really a problem either (as that will typically just _display_ things) -- monitor-specific `doWhatever` methods called from these views are. And currently any in plugins are protected automatically even without having to implement permission checks.

https://jenkins.io/doc/developer/extensions/jenkins-core/#administrativemonitor are just the ones we've indexed from public plugins.

Reply:
(missed this reply somehow 😢 )

I see, I've had a look and tried to change it back to administer, 
unfortunately without the change there's a big section up the top that  looks like this:
![image](https://user-images.githubusercontent.com/21194782/65752233-44afc700-e104-11e9-8ce6-5f33fe2dd059.png)

The only straightforward method I can see without updating all the admin monitors, is to not show admin monitors to users with the read permission.

![image](https://user-images.githubusercontent.com/21194782/65752658-23030f80-e105-11e9-937a-b8dd81b8b226.png)

Any thoughts on the approaches

Reply:
@daniel-beck addressed in 6cc6ba2a79c9fdd619a80b4a8a4f3575bd605eaa
by hiding admin monitors for system read permission, not ideal but I can't see a better way, open to ideas though